mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -109,6 +112,28 @@ public boolean allowsTransactionEmulation() {
 
 	@Override
 	public void start() {
+		if ( config.getNativeClientResource() == null ) {
+			startClientAndExtractDatabase();
+		}
+		else {
+			lookupDatabase();
+		}
+	}
+
+	private void lookupDatabase() {
+		try {
+			log.tracef( "Retrieving MongoDatabase from JNDI at %1$s", config.getNativeClientResource() );
+			mongoDb = (MongoDatabase) jndiService.locate( config.getNativeClientResource() );
+		}
+		catch (RuntimeException e) {
+			throw log.errorOnFetchJndiClientProperty( config.getNativeClientResource() );
+		}
+
+		// clear resources
+		this.jndiService = null;
```

Comment:
Not critical but I find it counter intuitive that the lookup method also does the clear, can we move it at the end of the start method?

Reply:
Sure I can